### ISSUE TITLE
std::unique_ptr not parsed properly, even with property compiler="cpp11" set

### ISSUE THREAD

**Issue by jkhoogland** [bytedeco/javacpp#169]
Hi,

I tried to distill from the documentation how to parse a c++ library with shared and unique ptrs, 
but it appears this is [not yet handled](https://groups.google.com/d/msg/javacpp-project/_WLOEbXcBCw/7Fsd5NWPAgAJ) by javacpp

I have code like the following (Foo.h)

```
#include <memory>
namespace a {
class FooI; // forward declaration
using FooU = std::unique_ptr<FooI>;
class FooI { /* actual declaration */ };
}
```
The javacpp parser gives the following type of error 
```
[ERROR] Failed to execute JavaCPP Builder: <...>/Foo.h:4: Could not parse declaration at '::'
```
Looking through various examples in the presets, I added the following details to the presets for the project:
```
@Properties(target="org.bytedeco.javacpp.foo", value={
        @Platform(compiler="cpp11",
                  include={"Foo.h"},
                  link={"f...@1.0.0"}))

public class foo implements InfoMapper {
    public void map(InfoMap infoMap) {
        infoMap.put(new Info("__cplusplus").define())
               .put(new Info("std::unique_ptr<a::FooI>").annotations("@UniquePtr").pointerTypes("Foo").define())
```
but this did not resolve the issue.

Would it be possible to add proper handling of this part of c++11 to javacpp ?

Much appreciated,
Jiri

**Comment by saudet** [comment]
Is `FooU` used anywhere in the public API that you need to access? Or is it not an issue if the type doesn't get mapped properly, as long as parsing doesn't fail?

**Comment by jkhoogland** [comment]
`FooU` is part of the public API.

**Comment by saudet** [comment]
I see. Well, it works exactly like a traditional `typedef`, so it shouldn't be too hard to get it working...

**Comment by saudet** [comment]
This should be fixed with the commit above. Let me know if you still have issues! Thanks

**Comment by jkhoogland** [comment]
will have a look, thanks

**Comment by saudet** [comment]
The fix is included in the newly released version 1.4. Thanks for reporting!
